File: code_samples/go/data_structures_tree.go
package main

import (
    "fmt"
)

// Data Structures: Tree
// AI/ML Training Sample

type Tree struct {
    Data string
}

func NewTree() *Tree {
    return &Tree{
        Data: "",
    }
}

func (s *Tree) Process(input string) {
    s.Data = input
}

func (s *Tree) Validate() bool {
    return len(s.Data) > 0
}

func (s *Tree) GetData() string {
    return s.Data
}

func main() {
    instance := NewTree()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/csharp/data_structures_tree.cs
using System;

/// <summary>
/// Data Structures: Tree
/// AI/ML Training Sample
/// </summary>
public class Tree
{
    private string data;
    
    public Tree()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Tree();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
